connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -405,13 +471,102 @@ protected ConfigDef config(Predicate<?> predicate) {
         return newDef;
     }
 
+    private static <T> ConverterDefaults converterDefaults(
+            Plugins plugins,
+            String connectorConverterConfig,
+            String workerConverterConfig,
+            String workerConverterVersionConfig,
+            Map<String, String> connectorProps,
+            Map<String, String> workerProps,
+            Class<T> converterType
+    ) {
+        /*
+        if a converter is specified in the connector config it overrides the worker config for the corresponding converter
+        otherwise the worker config is used, hence if the converter is not provided in the connector config, the default
+        is the one provided in the worker config
+
+        for converters which version is used depends on a several factors with multi-versioning support
+        A. If the converter class is provided as part of the connector properties
+            1. if the version is not provided,
+                - if the converter is packaged with the connector then, the packaged version is used
+                - if the converter is not packaged with the connector, the latest version is used
+            2. if the version is provided, the provided version is used
+        B. If the converter class is not provided as part of the connector properties, but provided as part of the worker properties
+            1. if the version is not provided, the latest version is used
+            2. if the version is provided, the provided version is used
+        C. If the converter class is not provided as part of the connector properties and not provided as part of the worker properties,
+        the converter to use is unknown hence no default version can be determined (null)
```

Comment:
The HeaderConverter has a default in WorkerConfig. The key and value converters do not have defaults.

...I wonder where the error appears when the WorkerConfig doesn't specify key/value converter...